Renaming TTCore namespace from mlir::tt -> mlir::tt::ttcore

sdjordjevicTT opened the pull request:
### Ticket
Closes https://github.com/tenstorrent/tt-mlir/issues/3863
Closes https://github.com/tenstorrent/tt-mlir/issues/3867

### Problem description
This issue is a part of the dialect namespace consolidation effort:
https://github.com/tenstorrent/tt-mlir/issues/3867

With the recent PR, we renamed the whole TT Dialect to TTCore, but we didn't update the corresponding **::mlir::tt** namespace:
https://github.com/tenstorrent/tt-mlir/pull/3772

### What's changed
Renaming TTCore mlir::tt namespace to mlir::tt::ttcore as we agreed in https://github.com/tenstorrent/tt-mlir/issues/3867.

### Checklist
- [x] New/Existing tests provide coverage for changes

sdjordjevicTT commented:
> Would you mind editing the PR description to reflect the _target_ ns structure we're shooting for? Surprised to see another massive renaming PR (and apparently not the last one, since there are still `tt` namespaces retained even in this one).

My bad for introducing the confusion. :(

I captured the required work for dialect namespace consolidation in this issue:
https://github.com/tenstorrent/tt-mlir/issues/3867

The reason for another massive renaming PR(and a few more) is that I would rather split things into a few logical changes rather than shooting for one big and complex change.

vroubtsovTT commented:
> I captured the required work for dialect namespace consolidation in this issue, and updated the PR description: #3867

You've answered my original query but from the ticket I see that the target structure is going to be 
we should consolidate the current set of dialect namespaces into:
```
::mlir::ttcore
::mlir::ttir
::mlir::ttnn
::mlir::ttkernel
::mlir::ttmetal
```
There is one aspect of this that is suboptimal: tenstorrent will no longer have a parent namespace like what `::mlir::tt` used to be. Such parent namespaces are quite useful for injecting c++ types/definitions common to sub-namespaces.

I'll add a comment to the issue, I think we should reconsider the change path before it's too late.

sdjordjevicTT commented:
> > I captured the required work for dialect namespace consolidation in this issue, and updated the PR description: #3867
> 
> You've answered my original query but from the ticket I see that the target structure is going to be
> 
> ```
> ::mlir::ttcore
> ::mlir::ttir
> ::mlir::ttnn
> ::mlir::ttkernel
> ::mlir::ttmetal
> ```
> 
> There is one aspect of this that is suboptimal: tenstorrent will no longer have a parent namespace like what `::mlir::tt` used to be. Such parent namespaces are quite useful for injecting c++ types/definitions common to sub-namespaces.
> 
> I'll add a comment to the issue, I think we should reconsider the change path before it's too late.
> 
> EDIT: comment added to associated ticket: [#3867 (comment)](https://github.com/tenstorrent/tt-mlir/issues/3867#issuecomment-2989186895)

Thanks for raising this, @vroubtsovTT. I will wait until we agree on the mentioned issue.

sdjordjevicTT commented:
Hi @vroubtsovTT, I updated the PR with the required changes as we agreed in this issue:
https://github.com/tenstorrent/tt-mlir/issues/3867#issuecomment-2989186895

I changed the ttcore dialect namespace from mlir::tt to mlir::tt::ttcore. :)

vprajapati-tt reviewed (approved):
Changes to CAPI/Bindings LGTM!
